packages/js/core/src/types/WrapError.ts

```diff
@@ -0,0 +1,212 @@
+import { CleanResolutionStep } from "../algorithms";
+
+/**
+Wrap error codes provide additional context to WrapErrors.
+
+Error code naming convention (approximate):
+    type of handler
+    type of functionality
+    piece of functionality
+    ==> handler_typeFn_pieceFn
+
+Error code map:
+  0-24 -> Client
+  25-49 -> URI resolution
+  50-74 -> Wrapper invocation & sub-invocation
+  75-255 -> Unallocated
+ */
+export enum WrapErrorCode {
```

Comment:
Some of these error names don't imply a negative situation, and could be interpreted positively. For example `CLIENT_GET_IMPLEMENTATIONS` doesn't inherently sound negative in nature. I'd recommend looking at some reference error codes (ex: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#client_error_responses) and seeing if there are better names we could use for these that imply the error scenario in question a bit more specifically. For example `CLIENT_VALIDATE_ABI` => `CLIENT_VALIDATE_ABI_NOT_FOUND`

Reply:
Good point. I changed them to the following:
```
  CLIENT_LOAD_WRAPPER_ERROR
  CLIENT_GET_FILE_ERROR,
  CLIENT_GET_IMPLEMENTATIONS_ERROR,
  CLIENT_VALIDATE_RESOLUTION_FAIL,
  CLIENT_VALIDATE_ABI_FAIL,
  CLIENT_VALIDATE_RECURSIVE_FAIL
```